bin/release_automation.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Check that Github CLI is installed
+command -v gh >/dev/null || { echo "Error: The Github CLI must be installed."; exit 1; }
+
+# Execute script commands from project's root directory
+SCRIPT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPT_PATH/.."
+
+# Check current branch is develop, master, or release/* branch
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+if [[ ! "$CURRENT_BRANCH" =~ "^develop$|^main$|^release/.*" ]]; then
+    echo "Releases should generally only be based on 'develop', 'main', or an earlier release branch."
+    echo "You are currently on the '$CURRENT_BRANCH' branch."
+    read -p "Are you sure you want to create a release branch from the '$CURRENT_BRANCH' branch? (y/n) " -n 1
+    echo ""
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+        printf "Aborting release...\n"
+        exit 1
+    fi
+fi
+
+# Confirm branch is clean
+[ -z "$(git status --porcelain)" ] || { git status; printf "\nUncommitted changes found. Aborting release script...\n"; exit 1; }
+
+# Ask for new version number
+CURRENT_VERSION_NUMBER=$(./node_modules/.bin/json -f package.json version)
+echo "Current Version Number:$CURRENT_VERSION_NUMBER"
+read -p "Enter the new version number: " VERSION_NUMBER
+
+# Insure javascript dependencies are up-to-date
+npm install || { echo "Error: 'yarn install' failed"; echo 1; }
```

Comment:
What do you think we should do about the `package-lock.json` change that results from running `npm install`? In theory it shouldn't happen, but it feels inevitable that it will happen again at some point. I just think we don't want to have the script blindly commit that change.

We could abort the script, but that seems like too much since we wouldn't want this to stop us from cutting a release branch. Feels like it might be better to do something like discard the `package-lock.json` change and, if there were changes, save a message to print at the end of the script.

Reply:
Are the changes happing inside the GB-Mobile repo or the GB repo?

Reply:
Anytime I run `npm install` I get this change to the `gutenberg-mobile/package-lock.json` file:

![image](https://user-images.githubusercontent.com/4656348/86178897-3a898180-baf7-11ea-93a1-eec70de7abf7.png)

My concern is not with this particular change, but with the possibility of the script committing other not-so-innocent changes to that file without us realizing it.

Reply:
Humm never saw that happening, could it be something specific in your machine setup?
We shouldn't see any changes after 'npm install' but if we noticed them  we probably should be able to revert them with a `git revert` ?

Reply:
> Humm never saw that happening, could it be something specific in your machine setup?

Quite possibly, but a number of people have seen it, so it's definitely not just me: p1592831812099400-slack-mobile-gutenberg-monorepo

> We shouldn't see any changes after 'npm install'...

`npm install` will update the [version line](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/package-lock.json#L3) of the `package-lock.json` file if it sees the version has changed, so we want to commit that change at some point in the script. Otherwise, every dev will get a diff on the `package-lock.json` file anytime they run `npm install`.

> ...but if we noticed them we probably should be able to revert them with a git revert ?

Do you mean after the script has finished, or as part of the script itself. My concern with doing it after the script has finished is that we wouldn't actually notice the change because I doubt anyone really does a thorough code review on the release PRs since they're huge even ignoring all the bundle changes. Doing it as part of the script itself seems like the better idea (i.e., only commit the update to the version line in the `package-lock.json`).

Actually, maybe the better approach is to do the `npm install` before we set the version, like we do now (and **not** move the `npm install` to after the version is set [like I suggested earlier](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2398#discussion_r444437234)). After `npm install` we could just reset the `package-lock.json` file. Then when we set the version, we could manually set it for both the `package.json` and `package-lock.json` files.

Reply:
We solved this by using `npm ci`